### PR TITLE
[expo-cli] turtle v2: create app before running build

### PR DESCRIPTION
**Pull request by dsokal**
For Turtle v2 builds, we want to create an app on Expo Servers before scheduling the build.

### CODE REVIEW

File: packages/expo-cli/src/commands/build-native/index.ts [hunk 1]
```diff
@@ -1,28 +1,46 @@
...
+  const builder = new Builder(ctx);
+  const result = await builder.getLatestBuildsAsync();
   printBuildTable(result.builds);
 }
 
+async function createBuilderContextAsync(projectDir: string): Promise<BuilderContext> {
+  const user: User = await UserManager.ensureLoggedInAsync();
+  const { exp } = getConfig(projectDir);
+  const accountName = exp.owner || user.username;
+  const projectName = exp.slug || 'untitled';
```
Comment (fson) [inline packages/expo-cli/src/commands/build-native/index.ts:34]: Are there any repercussions from creating an app named "untitled" on the server (e.g. how hard is it for the user to rename it afterwards? What happens if they already have a different app named `untitled`?). Would it make sense to ask them to give a name if we can't find the name?

File: packages/expo-cli/src/commands/build-native/Builder.ts [hunk 2]
```diff
@@ -37,62 +32,71 @@ interface PresignedPost {
   fields: object;
 }
 
+export interface BuilderContext {
+  projectDir: string;
+  user: User;
+  accountName: string;
```
[the remark this answers is not in the document]
Comment (dsokal) [reply packages/expo-cli/src/commands/build-native/Builder.ts:38]: https://exponent-internal.slack.com/archives/C3M7V1T7Z/p1592212385188000

[the remark this answers is not in the document]
Comment (dsokal) [reply packages/expo-cli/src/commands/build-native/Builder.ts:38]: I'd say that we should merge it for now (as the PR introducing the new endpoint in www is already merged).

File: packages/expo-cli/src/commands/build-native/index.ts (hunk 1 above)
Comment (dsokal) [reply packages/expo-cli/src/commands/build-native/index.ts]: I sent you a more detailed answer on Slack. I made expo-cli throw an error if `slug` is empty.

Comment (fson) [reply packages/expo-cli/src/commands/build-native/index.ts]: I've fixed the type of `slug` in `ExpoConfig` here: https://github.com/expo/expo-cli/pull/2260.

**Pull request merged**